api/server/runner_fninvoke.go

```diff
@@ -15,6 +16,26 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+var (
+	bufPool = &sync.Pool{New: func() interface{} { return new(bytes.Buffer) }}
+)
+
+var _ http.ResponseWriter = new(syncResponseWriter)
+
+// implements http.ResponseWriter
+// this little guy buffers responses from user containers and lets them still
+// set headers and such without us risking writing partial output [as much, the
+// server could still die while we're copying the buffer]. this lets us set
+// content length and content type nicely, as a bonus. it is sad, yes.
+type syncResponseWriter struct {
+	headers http.Header
```

Comment:
A separate concern to this pr.

Reply:
if this is a behavior change we should change it back I guess and do it separately (since this PR should only concern removal of v1 api), it's unclear / i don't know off hand writing this comment

Reply:
Yeah for buffering I've left it as is,